packages/x-date-pickers/src/internals/components/CalendarOrClockPicker/useFocusManagement.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+interface FocusState {
+  focusedPicker: 'year' | 'month' | 'day';
+  hasFocus: boolean;
```

Comment:
Nitpick: Wouldn't having `focusedPicker: 'year' | 'month' | 'day' | undefined` without `hasFocus` flag be easier to manage and still achieve the same functionality? 🤔

Reply:
Yes, it would work too with `hasFocus = focusedPicker !== undefined` but not sure it's would be much easier.

However, we can have `focusedPicker: 'year' | 'month' | 'day' | null` and instead of using `hasFocus`, use `focusedPicker` for <Calendar />, such that 

```jsx
<YearPicker
	hasFocus={focusedPicker!==undefined ? focusedPicker==='year' : undefined}
	...
/>
<MonthPicker
	hasFocus={focusedPicker!==undefined ? focusedPicker==='month' : undefined}
	...
/>
<DayPicker
	hasFocus={focusedPicker!==undefined ? focusedPicker==='day' : undefined}
	...
/>
```

Reply:
Yup, or `null`, whichever would be easiest to implement, but I meant that from the code it doesn't seem that separate `hasFocus` flag add much in flexibility 🤔 
```tsx
<YearPicker hasFocus={focusedPicker === 'year'} />
<MonthPicker hasFocus={focusedPicker === 'month'} />
<DayPicker hasFocus={focusedPicker === 'day'} />
```
But in any case, it's just food for thought as eventually we might see that to achieve certain state/behaviour we might need a separate flag. 😄